aks/storage_account/resources.tf

```diff
@@ -0,0 +1,153 @@
+locals {
+  name = var.name != null ? var.name : "default"
+  storage_account_name = var.storage_account_name_override != null ? var.storage_account_name_override : lower(
+    "${var.azure_resource_prefix}${var.service_short}${local.name}${var.config_short}sa"
+  )
+
+  # Determine replication type based on environment
+  effective_replication_type = var.use_production_replication ? var.production_replication_type : var.account_replication_type
+
+  default_tags = {
+    Service     = var.service_name
+    Environment = var.environment
+  }
+  tags = merge(local.default_tags, var.tags)
+
+  # Default management policy for deleting blobs after 7 days
+  default_management_policy_rule = {
+    name    = "DeleteAfter7Days"
+    enabled = true
+    filters = {
+      blob_types = ["blockBlob"]
+    }
+    actions = {
+      base_blob = {
+        delete_after_days_since_modification_greater_than = 7
+      }
+    }
+  }
+
+  # Combine default policy with custom policies if default_management_policy is true
+  management_policy_rules = var.default_management_policy ? concat([local.default_management_policy_rule], var.management_policy_rules) : var.management_policy_rules
+}
+
+data "azurerm_resource_group" "main" {
+  count = var.use_azure ? 1 : 0
+  name  = "${var.azure_resource_prefix}-${var.service_short}-${var.config_short}-rg"
+}
+
+resource "azurerm_storage_account" "main" {
+  count = var.use_azure && var.create_storage_account ? 1 : 0
```

Comment:
I don't think we need any count settings here, and probably not on any of the other resources.
You would decide to use the module or not at the calling end and create a single storage account.